jetty-ee10/jetty-ee10-tests/jetty-ee10-test-http2-webapp/src/test/java/org/eclipse/jetty/test/webapp/StressHTTP2Test.java

```diff
@@ -0,0 +1,729 @@
+                        {
+                            // ignore
+                        }
+
+                        // if ((j + 1) % 100 == 0)
+                        //     System.err.println(Thread.currentThread().getName() + " processed " + (j + 1));
+                    }
+                }
+                return null;
+            });
+            futures.add(f);
+        }
+
+        for (Future<?> future : futures)
+        {
+            future.get();
+        }
+
+        // Assert that no thread is stuck in WAITING state, i.e.: blocked on some lock.
+        await().atMost(30, TimeUnit.SECONDS).until(() ->
+        {
+            TestQueuedThreadPool queuedThreadPool = server.getBean(TestQueuedThreadPool.class);
+            for (Thread thread : queuedThreadPool.getCreatedThreads())
+            {
+                Thread.State state = thread.getState();
+                if (state == Thread.State.WAITING)
+                    return false;
+            }
+            return true;
+        });
+    }
+
+    @BeforeEach
+    public void setUp() throws Exception
+    {
+        startServer();
+        executorService = Executors.newFixedThreadPool(N_THREADS);
+    }
+
+    @AfterEach
+    public void tearDown()
+    {
+        executorService.shutdownNow();
+        LifeCycle.stop(server);
+    }
+
+    private void startServer() throws Exception
+    {
+        QueuedThreadPool qtp = new TestQueuedThreadPool();
+        server = new Server(qtp);
+
+        HttpConfiguration httpConfiguration = new HttpConfiguration();
+        httpConfiguration.setOutputBufferSize(1);
+        ConnectionFactory connectionFactory = new HTTP2CServerConnectionFactory(httpConfiguration);
+        ServerConnector serverConnector = new ServerConnector(server, connectionFactory);
+        serverConnector.setPort(0);
+        server.addConnector(serverConnector);
+
+        ServletContextHandler targetContextHandler = new ServletContextHandler();
+        targetContextHandler.setContextPath("/");
+        targetContextHandler.addServlet(new SyncEE10Servlet(), "/*");
+
+        server.setHandler(targetContextHandler);
+
+        server.start();
+    }
+
+    private static class TestQueuedThreadPool extends QueuedThreadPool
+    {
+        private static final List<Thread> CREATED_THREADS = new CopyOnWriteArrayList<>();
+        private static final ThreadFactory THREAD_FACTORY = r ->
+        {
+            Thread thread = new Thread(r);
+            CREATED_THREADS.add(thread);
+            return thread;
+        };
+
+        public TestQueuedThreadPool()
+        {
+            super(N_THREADS * 2, 8, 60000, -1, null, null, THREAD_FACTORY);
+        }
+
+        public List<Thread> getCreatedThreads()
+        {
+            return CREATED_THREADS;
+        }
+    }
+
+    private static class SyncEE10Servlet extends HttpServlet
+    {
+        private static final byte[] DATA = """
+            <?xml version="1.0" encoding="UTF-8"?>
```

Comment:
It is confusing to have one of our XML files as the data.  This will date and get even more confusing.
Why not just random data?

Reply:
I wanted data that was humanly readable to be able to peek at the buffers, but that XML could be replaced by a repeated lorem ipsum instead.